moveit_planners/ompl/ompl_interface/test/test_state_validity_checker.cpp

```diff
@@ -0,0 +1,361 @@
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Jeroen De Maeyer */
+/**
+ *    This test checks the basics of a StateValidityChecker:
+ *        - Can we create one?
+ *        - States inside and outside joint limits.
+ *        - States that are in self-collision.
+ *
+ *    It does not yet test:
+ *        - Collision with objects in the environment.
+ *        - Any type of path constraints.
+ *
+ *    The test do show what is minimally required to create a working StateValidityChecker.
+ **/
+
+#include "load_test_robot.h"
+
+#include <limits>
+#include <ostream>
+
+#include <gtest/gtest.h>
+
+#include <moveit/ompl_interface/detail/state_validity_checker.h>
+#include <moveit/ompl_interface/model_based_planning_context.h>
+#include <moveit/ompl_interface/parameterization/joint_space/joint_model_state_space.h>
+#include <moveit/planning_scene/planning_scene.h>
+
+#include <ompl/geometric/SimpleSetup.h>
+
+/** \brief Use this flag to turn on extra output on std::cout for debugging. **/
+constexpr bool VERBOSE{ false };
+
+/** \brief Pretty print std:vectors **/
+std::ostream& operator<<(std::ostream& os, const std::vector<double>& v)
+{
+  os << "( ";
+  for (auto value : v)
+    os << value << ", ";
+  os << " )";
+  return os;
+}
+
+/** \brief Generic implementation of the tests that can be executed on different robots. **/
+class TestStateValidityChecker : public ompl_interface_testing::LoadTestRobot, public testing::Test
+{
+public:
+  TestStateValidityChecker(const std::string& robot_name, const std::string& group_name)
+    : LoadTestRobot(robot_name, group_name)
+  {
+  }
+
+  // /***************************************************************************
+  //  * START Test implementations
+  //  * ************************************************************************/
+
+  void testConstructor()
+  {
+    ompl::base::StateValidityCheckerPtr checker =
+        std::make_shared<ompl_interface::StateValidityChecker>(planning_context_.get());
+  }
+
+  /** This test takes a state that is inside the joint limits and collision free as input. **/
+  void testJointLimits(const std::vector<double>& position_in_limits)
+  {
+    // create a validity checker for this test
+    auto checker = std::make_shared<ompl_interface::StateValidityChecker>(planning_context_.get());
+    checker->setVerbose(VERBOSE);
+
+    robot_state_->setJointGroupPositions(joint_model_group_, position_in_limits);
+
+    // use a scoped OMPL state so we don't have to call allocState and freeState
+    // (as recommended in the OMPL documantion)
+    ompl::base::ScopedState<> ompl_state(state_space_);
+    state_space_->copyToOMPLState(ompl_state.get(), *robot_state_);
+
+    if (VERBOSE)
+      ROS_INFO_STREAM(ompl_state.reals());
```

Comment:
You could just change this to `ROS_DEBUG_STREAM_NAMED("test_joint_limits", ompl_state.reals());` and then it will only show up in the output when you have debug logging enabled.

Reply:
done